src/platformio.ts

```diff
@@ -100,6 +120,37 @@ export class PlatformIO {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private configureAutoUpdateIncludes(): void {
+        const autoUpdateIncludes = vscode.workspace.getConfiguration("platformio").get<boolean>("autoUpdateIncludes");
+        if (autoUpdateIncludes && !this._autoUpdateIncludesEnabled) {
+            const fileSystemWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**");
+            this._autoUpdateIncludesDisposables.push(fileSystemWatcher);
+            fileSystemWatcher.onDidCreate((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            fileSystemWatcher.onDidDelete((e: vscode.Uri) => this.fileSystemDidChange(e), this, this._autoUpdateIncludesDisposables);
+            const iniFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher("**/platformio.ini");
+            this._autoUpdateIncludesDisposables.push(iniFileWatcher);
+            iniFileWatcher.onDidChange((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+            iniFileWatcher.onDidCreate((e: vscode.Uri) => this.addIncludePath(true), this, this._autoUpdateIncludesDisposables);
+        }

```

Comment:
That is a good idea. Do you think we should still filter on the file extension in the lib folder to avoid updates when the user adds files to his "own" libraries?